runtime/backend/options.h

```diff
@@ -0,0 +1,217 @@
+#include <executorch/runtime/core/error.h>
+#include <executorch/runtime/core/span.h>
+#include <array>
+#include <cstddef>
+#include <cstring>
+#include <variant>
+
+namespace executorch {
+namespace runtime {
+
+static constexpr size_t kMaxOptionKeyLength = 64;
+static constexpr size_t kMaxOptionValueLength = 256;
+
+// All string keys and values must have static storage duration (string
+// literals, static const char arrays, or global constants). The BackendOptions
+// class does NOT take ownership of strings.
+using OptionValue =
+    std::variant<bool, int, std::array<char, kMaxOptionValueLength>>;
+
+struct BackendOption {
+  // key is the name of the backend option, like num_threads, enable_profiling,
+  // etc
+  char key[kMaxOptionKeyLength];
+  // value is the value of the backend option, like 4, true, etc
+  OptionValue value;
+};
+
+/**
+ * A template class for storing and managing backend-specific configuration
+ * options.
+ *
+ * This class provides a type-safe way to store key-value pairs for backend
+ * configuration, with compile-time capacity limits and runtime type checking.
+ * It supports bool, int, and const char* value types.
+ *
+ * @tparam MaxCapacity The maximum number of options that can be stored
+ */
+template <size_t MaxCapacity>
+class BackendOptions {
+ public:
+  /**
+   * Copy constructor
+   */
+  BackendOptions(const BackendOptions& other) : size_(other.size_) {
+    for (size_t i = 0; i < size_; ++i) {
+      options_[i] = other.options_[i];
+    }
+  }
+
+  /**
+   * Copy assignment operator
+   */
+  BackendOptions& operator=(const BackendOptions& other) {
+    if (this != &other) {
+      size_ = other.size_;
+      for (size_t i = 0; i < size_; ++i) {
+        options_[i] = other.options_[i];
+      }
+    }
+    return *this;
+  }
+
+  /**
+   * Default constructor - initializes with zero options.
+   */
+  BackendOptions() : size_(0) {}
+
+  /**
+   * Returns a const view of all stored options as a Span.
+   *
+   * @return A const Span containing all BackendOption entries
+   */
+  executorch::runtime::Span<const BackendOption> view() const {
+    return executorch::runtime::Span<const BackendOption>(options_, size_);
+  }
+
+  /**
+   * Returns a mutable view of all stored options as a Span.
+   *
+   * @return A mutable Span containing all BackendOption entries
+   */
+  executorch::runtime::Span<BackendOption> mutable_view() {
+    return executorch::runtime::Span<BackendOption>(options_, size_);
+  }
+
+  /**
+   * Sets a boolean option value for the given key.
+   * If the key already exists, updates its value. Otherwise, adds a new option.
+   *
+   * @tparam N The length of the key string (automatically deduced)
+   * @param key The option key (must be a string literal or array)
+   * @param value The boolean value to set
+   * @return Error::Ok on success, Error::InvalidArgument if storage is full
+   */
+  template <size_t N>
+  Error set_option(const char (&key)[N], bool value) noexcept {
```

Comment:
Do we need N? It will just create N different methods for many possible lengths. My suggestion is to remove N as template param and expect callsites to be something like
```
char my_option_name[kMaxOptionKeyLength] = "use_gpu";
...set_option(my_option_name, True)
```